test/drop_test/action.py

```diff
@@ -324,3 +326,20 @@ async def run(self, drop: ffi.Drop):
 
     def __str__(self):
         return f"CompareTrees({self._out_dir}, {self._tree})"
+
+
+class WaitForResume(Action):
+    def __init__(self, uuid_slot: int, file_id: str, tmp_file_path: str):
+        self._uuid_slot = uuid_slot
+        self._file_id = file_id
+        self._tmp_file_path = tmp_file_path
+
+    async def run(self, drop: ffi.Drop):
+        stat = os.stat(self._tmp_file_path)
+        await drop._events.wait_for(event.Start(self._uuid_slot, self._file_id), False)
+        await drop._events.wait_for(
+            event.Progress(self._uuid_slot, self._file_id, stat.st_size), False
+        )
+
+    def __str__(self):
+        return f"WaitForResume({print_uuid(self._uuid_slot)}, {self._file_id}, {self._tmp_file_path})"
```

Comment:
I've tried that but it was simply failing. There is a possibility of race, because if we wait for `Start` and then wait for `Progress` we need to read the file size in between, but the file might be already changed and the `Progress` event already pending before we will wait for it. The solution is to read the file size before `Start`.